applicationinsights/django/middleware.py

```diff
@@ -0,0 +1,266 @@
+            'django.contrib.sessions.middleware.SessionMiddleware',
+            'django.middleware.common.CommonMiddleware',
+            'django.middleware.csrf.CsrfViewMiddleware',
+            'django.contrib.auth.middleware.AuthenticationMiddleware',
+            'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
+            'django.contrib.messages.middleware.MessageMiddleware',
+            'django.middleware.clickjacking.XFrameOptionsMiddleware',
+            # ... or whatever is above for you ...
+            'applicationinsights.django.ApplicationInsightsMiddleware',   # Add this middleware to the end
+        ]
+        
+        # If on Django >= 1.10
+        MIDDLEWARE = [
+            # ... or whatever is below for you ...
+            'django.middleware.security.SecurityMiddleware',
+            'django.contrib.sessions.middleware.SessionMiddleware',
+            'django.middleware.common.CommonMiddleware',
+            'django.middleware.csrf.CsrfViewMiddleware',
+            'django.contrib.auth.middleware.AuthenticationMiddleware',
+            'django.contrib.messages.middleware.MessageMiddleware',
+            'django.middleware.clickjacking.XFrameOptionsMiddleware',
+            # ... or whatever is above for you ...
+            'applicationinsights.django.ApplicationInsightsMiddleware',   # Add this middleware to the end
+        ]
+    
+    And then, add the following to your settings.py file:
+    
+    .. code:: python
+    
+        APPLICATION_INSIGHTS = {
+            # (required) Your Application Insights instrumentation key
+            'ikey': "00000000-0000-0000-0000-000000000000",
+            
+            # (optional) By default, if DEBUG is True, then the middleware will
+            # not log any events.  To override this behavior, set debug_ikey, below:
+            'debug_ikey': "00000000-0000-0000-0000-000000000000",
+            
+            # (optional) By default, request names are logged as the fully-qualified
+            # name of the view.  To disable this behavior, specify:
+            'use_operation_url': True,
+            
+            # (optional) By default, arguments to views are tracked as custom
+            # properties.  To disable this, specify:
+            'record_view_arguments': False,
+            
+            # (optional) Events are submitted to Application Insights asynchronously.
+            # send_interval specifies how often the queue is checked for items to submit.
+            # send_time specifies how long the sender waits for new input before recycling
+            # the background thread.
+            'send_interval': 1.0, # Check every second
+            'send_time': 3.0, # Wait up to 3 seconds for an event
+            
+            # (optional, uncommon) If you must send to an endpoint other than the
+            # default endpoint, specify it here:
+            'endpoint': "https://dc.services.visualstudio.com/v2/track",
```

Comment:
This change wouldn't affect any SDK functionality, it's just a different domain that points to the same place